frontend/src/components/panels/Document.vue

```diff
@@ -379,19 +379,29 @@ export default defineComponent({
 			this.canvasCursor = updateMouseCursor.cursor;
 		});
 		this.editor.dispatcher.subscribeJsMessage(TriggerTextCommit, () => {
-			if (this.textInput) this.editor.instance.on_change_text(this.textInput.textContent || "");
+			if (this.textInput) {
```

Comment:
Please add a comment explaining the motivation and details for this since it'll probably be confusing to people in the future.